### DISCUSSION TITLE
Does deleted items from Maintenance go to recycle bin or are they permanently deleted?

### DISCUSSION THREAD

**Discussion by lamyergeier** [stashapp/stash#4758] [General]
Does deleted items from Maintenance go to recycle bin or are they permanently deleted? May be we can have a recycle-bin folder for stash and keep all deleted stuff there?

**Comment by DogmaDragon** [comment]
`Clean` task only removes the data from the database, not files. 
`Clean generated files` permanently deletes the generated files.
`Delete` action offers a prompt to either delete the metadata or also permanently delete the attached file.

**Reply by lamyergeier** [reply]
@DogmaDragon 

May be `Clean generated files` can be put in a recycle bin folder for Stash? (to be able to undo costly mistakes)

**Reply by DogmaDragon** [reply]
I'm not sure if there are technical reasons it's not done this way, but you make a feature request https://github.com/stashapp/stash/issues/new/choose.

**Comment by feederbox826** [comment]
Converted into an issue https://github.com/stashapp/stash/issues/4949
